### DISCUSSION TITLE
Input different elasticity tensor Cijkl in tensor_mechanics

### DISCUSSION THREAD

**Discussion by hugy888** [idaholab/moose#21486] [Q&A Modules: Solid mechanics] [answered]
Hello,

We decided to calculate elastic properties of multiphase polycrystals using tensor_mechanics module. 
I found from test examples that Cijkl for single phase polycrystal could be input within Materials object as followed:
```
  [./elasticity_tensor]
    type = ComputeElasticityTensorCP
    C_ijkl = '1.684e5 1.214e5 1.214e5 1.684e5 1.214e5 1.684e5 0.754e5 0.754e5 0.754e5'
    fill_method = symmetric9
    read_prop_user_object = prop_read
  [../]
```

I'm wondering there's a way that I could input different C_ijkls for different phases or elements? Similar to what we did to input different euler angles. Thanks.

```
[UserObjects]
  [./prop_read]
    type = PropertyReadFile
    prop_file_name = 'euler_ang_file.txt'
    # Enter file data as prop#1, prop#2, .., prop#nprop
    nprop = 3
    read_type = block
    nblock= 2
  [../]
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

It would not be very hard to do but it's not present in the current code afaik. You can inspire yourself from the treatment of the euler angle with a PropertyReadFile. 

I ll let others comment on what spatial granularity is appropriate for the stiffness tensor. 

Guillaume
